cellar_rebalancer/src/cellars/uniswapv3.rs

```diff
@@ -158,11 +166,42 @@ impl<T: 'static + Middleware> UniswapV3CellarState<T> {
     }
 }
 
-// Struct for UniswapV3CellarTickInfo
-#[derive(Clone, Debug)]
-pub struct UniswapV3CellarTickInfo {
-    pub(crate) token_id: U256,
-    pub(crate) tick_upper: i32,
-    pub(crate) tick_lower: i32,
-    pub(crate) weight: u32,
+pub struct UniswapV3CellarAllocator;
+
+#[async_trait]
+impl server::UniswapV3CellarAllocator for UniswapV3CellarAllocator {
+    async fn rebalance(
+        &self,
+        request: tonic::Request<RebalanceRequest>,
+    ) -> Result<tonic::Response<RebalanceResponse>, tonic::Status> {
+        let request = request.get_ref();
+        let tick_ranges: Vec<proto::TickRange> = request.data
+            .clone()
+            .into_iter()
+            .map(|d| {
+                proto::TickRange {
+                    upper: d.upper_price,
+                    lower: d.lower_price,
+                    weight: d.weight
+                }
+            })
+            .collect();
+        let pair_id = APP.config()
+            .cellars
+            // sketchy
+            .get(0)
+            .unwrap()
+            .pair_id;
+        let eth_gas_price = match cellars::get_gas_price().await {
+            Ok(gp) => gp,
+            Err(err) => {
+                error!("Failed to get cellar gas price: {:?}", err);
+                // what's the right way to handle this?
+                0.into()
+            }
+        };
+
+        allocation::decide_rebalance(tick_ranges, pair_id, eth_gas_price.as_u64()).await;
```

Comment:
I wonder if we should await on this or tokio spawn this into a new thread and return.

Errors in the allocation module aren't really the client's responsibility to fix

Reply:
Pushed a fix

Reply:
As a counterpoint, clients of most HTTP services can't really do anything about getting 500-level error codes back but it is useful information to know that the endpoint is broken.

Reply:
I think it's useful to be able to return an error if the data doesn't parse or is otherwise invalid.